### PR TITLE
MAINT: update OMMBV usage and install

### LINKED ISSUE #63: TST/BUG: OMMBV requires manual install on GitHub Actions
**Describe the bug**
OMMBV does not install from pip. Some sort of numpy conflict.

**To Reproduce**
See logs here 
https://github.com/pysat/pysatMissions/actions/runs/1099735603

Currently using --no-binary flag to install separately,  likely needs updates at main package.  Documented there at https://github.com/rstoneback/OMMBV/issues/50

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Closes #63.  Updates usage of OMMBV.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

via GA

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes -- fixes for previous pull in same version

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -42,7 +42,7 @@ install_requires =
   apexpy
   geospacepy
   numpy
-  OMMBV
+  OMMBV>1.0
```
Comment (jklenzing) [inline setup.cfg:45]:
```suggestion
  OMMBV>=1.0
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -11,24 +11,36 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.7, 3.8, 3.9]
+        python-version: ["3.9", "3.10"]
```
Comment (jklenzing) [inline .github/workflows/main.yml:14]:
```suggestion
        python-version: ["3.8", "3.9", "3.10"]
```

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -11,24 +11,36 @@ jobs:
...
         os: [ubuntu-latest]
+        numpy_ver: ["latest"]
+        include:
+          - python-version: "3.8"
+            numpy_ver: "latest"
+            os: ubuntu-latest
```
Comment (jklenzing) [inline .github/workflows/main.yml:20]:
```suggestion
```

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -11,24 +11,36 @@ jobs:
...
+
+    - name: Install requirements for testing setup
       run: |
         python -m pip install --upgrade pip
         pip install -r test_requirements.txt
+
+    - name: Install NEP29 dependencies
+      if: ${{ matrix.numpy_ver != 'latest'}}
+      run: |
+        pip install --no-binary :numpy: numpy==${{ matrix.numpy_ver }}
```
Comment (jklenzing) [inline .github/workflows/main.yml:39]:
```suggestion
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -11,6 +11,7 @@ This project adheres to [Semantic Versioning](https://semver.org/).
   * Improve docstrings throughout
 * Testing
   * Add style check for docstrings
+  * Include support for testing against multiple versions of numpy
```
Comment (jklenzing) [inline CHANGELOG.md:14]:
```suggestion
```

File: pysatMissions/methods/spacecraft.py [hunk 6]
```diff
@@ -173,7 +173,7 @@ def project_ecef_vector_onto_sc(inst, x_label, y_label, z_label,
 
     # TODO(#65): add checks for existence of ecef labels in inst
 
-    x, y, z = OMMBV.project_ecef_vector_onto_basis(
+    x, y, z = OMMBV.project_ECEF_vector_onto_basis(
```
Comment (jklenzing) [inline pysatMissions/methods/spacecraft.py:176]: note this changed in ommbv 1.0

Comment (rstoneback) [inline pysatMissions/methods/spacecraft.py:176]:
```suggestion
    x, y, z = OMMBV.vector.project_onto_basis(
```
The `project_ECEF_vector_onto_basis` change was apparently a typo on my part when I was moving and deprecating the call.

File: pysatMissions/methods/spacecraft.py [hunk 7]
```diff
@@ -38,45 +38,45 @@ def add_ram_pointing_sc_attitude_vectors(inst):
...
     # Begin with z along Nadir (towards Earth)
     # if orbit isn't perfectly circular, then the s/c z vector won't
     # point exactly along nadir. However, nadir pointing is close enough
     # to the true z (in the orbital plane) that we can use it to get y,
     # and use x and y to get the real z
     inst['sc_zhat_ecef_x'], inst['sc_zhat_ecef_y'], inst['sc_zhat_ecef_z'] = \
-        OMMBV.normalize_vector(-inst['position_ecef_x'],
-                               -inst['position_ecef_y'],
-                               -inst['position_ecef_z'])
+        OMMBV.vector.normalize_vector(-inst['position_ecef_x'],
```
Comment (rstoneback) [inline pysatMissions/methods/spacecraft.py:51]:
```suggestion
        OMMBV.vector.normalize(-inst['position_ecef_x'],
```

File: pysatMissions/methods/spacecraft.py [hunk 8]
```diff
@@ -38,45 +38,45 @@ def add_ram_pointing_sc_attitude_vectors(inst):
...
+                                   inst['sc_zhat_ecef_z'],
+                                   inst['sc_xhat_ecef_x'],
+                                   inst['sc_xhat_ecef_y'],
+                                   inst['sc_xhat_ecef_z'])
     # Normalize since Xhat and Zhat from above may not be orthogonal
     inst['sc_yhat_ecef_x'], inst['sc_yhat_ecef_y'], inst['sc_yhat_ecef_z'] = \
-        OMMBV.normalize_vector(inst['sc_yhat_ecef_x'],
-                               inst['sc_yhat_ecef_y'],
-                               inst['sc_yhat_ecef_z'])
+        OMMBV.vector.normalize_vector(inst['sc_yhat_ecef_x'],
```
Comment (rstoneback) [inline pysatMissions/methods/spacecraft.py:66]:
```suggestion
        OMMBV.vector.normalize(inst['sc_yhat_ecef_x'],
```

**Review by rstoneback — APPROVED** [review]

**Pull request merged**